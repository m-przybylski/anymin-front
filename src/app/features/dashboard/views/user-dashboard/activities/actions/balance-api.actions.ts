// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { MoneyDto } from '@anymind-ng/api';

export enum BalanceApiActionTypes {
  LoadExpertBalanceSuccess = '[Balance API] Load Balance Success',
  LoadExpertBalanceFailure = '[Balance API] Load Balance Failure',
}

export class LoadExpertBalanceSuccessAction implements Action {
  public readonly type = BalanceApiActionTypes.LoadExpertBalanceSuccess;

  constructor(public payload: MoneyDto) {}
}

export class LoadExpertBalanceFailureAction implements Action {
  public readonly type = BalanceApiActionTypes.LoadExpertBalanceFailure;

  // tslint:disable-next-line:no-any
  constructor(public payload: any) {}
}

export type BalanceApiActionUnion = LoadExpertBalanceSuccessAction | LoadExpertBalanceFailureAction;
