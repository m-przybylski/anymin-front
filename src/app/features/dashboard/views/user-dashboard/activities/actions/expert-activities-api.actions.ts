// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { IActivitiesData } from '../services/expert-activities.service';
import { GetProfileActivities, GetProfileActivity } from '@anymind-ng/api';

export enum ExpertActivitiesApiActionTypes {
  LoadExpertActivitiesSuccess = '[Activities API] Load Expert Activities Success',
  LoadExpertActivitiesFailure = '[Activities API] Load Expert Activities Failure',
  LoadExpertActivitiesWithImportantSuccess = '[Activities API] Load Expert All Activities Success',
  LoadExpertActivitiesWithImportantFailure = '[Activities API] Load Expert All Activities Failure',
  LoadExpertActivitySuccess = '[Activities API] Load Expert Activity Success',
  LoadExpertActivityFailure = '[Activities API] Load Expert Activity Failure',
}

export class LoadExpertActivitiesWithImportantSuccessAction implements Action {
  public readonly type = ExpertActivitiesApiActionTypes.LoadExpertActivitiesWithImportantSuccess;

  constructor(public payload: IActivitiesData) {}
}

export class LoadExpertActivitiesWithImportantFailureAction implements Action {
  public readonly type = ExpertActivitiesApiActionTypes.LoadExpertActivitiesWithImportantFailure;

  // tslint:disable-next-line:no-any
  constructor(public payload: any) {}
}

export class LoadExpertActivitiesSuccessAction implements Action {
  public readonly type = ExpertActivitiesApiActionTypes.LoadExpertActivitiesSuccess;

  constructor(public payload: GetProfileActivities) {}
}

export class LoadExpertActivitiesFailureAction implements Action {
  public readonly type = ExpertActivitiesApiActionTypes.LoadExpertActivitiesFailure;

  // tslint:disable-next-line:no-any
  constructor(public payload: any) {}
}

export class LoadExpertActivitySuccessAction implements Action {
  public readonly type = ExpertActivitiesApiActionTypes.LoadExpertActivitySuccess;

  constructor(public payload: GetProfileActivity) {}
}

export class LoadExpertActivityFailureAction implements Action {
  public readonly type = ExpertActivitiesApiActionTypes.LoadExpertActivityFailure;

  // tslint:disable-next-line:no-any
  constructor(public payload: any) {}
}
export type ExpertActivitiesApiActionUnion =
  | LoadExpertActivitiesWithImportantSuccessAction
  | LoadExpertActivitiesWithImportantFailureAction
  | LoadExpertActivitiesSuccessAction
  | LoadExpertActivitiesFailureAction
  | LoadExpertActivitySuccessAction
  | LoadExpertActivityFailureAction;
