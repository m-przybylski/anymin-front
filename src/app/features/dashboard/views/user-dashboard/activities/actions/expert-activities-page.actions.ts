// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { GetProfileActivity } from '@anymind-ng/api';

export enum ExpertActivitiesPageActionTypes {
  LoadMoreExpertActivities = '[Activities Page] Load More Expert Activities',
  ShowImportantActivities = '[Activities Page] Show Important Activities',
  HideImportantActivities = '[Activities Page] Hide Important Activities',
  ActivityDetailsClosed = '[Activities Page] Activity Details Closed',
  ActivityRowClick = '[Activity Page] Activity Row Click',
}

export class LoadMoreExpertActivitiesAction implements Action {
  public readonly type = ExpertActivitiesPageActionTypes.LoadMoreExpertActivities;

  constructor(public payload: { currentOffset: number; offsetIterator: number }) {}
}

export class ShowImportantActivitiesAction implements Action {
  public readonly type = ExpertActivitiesPageActionTypes.ShowImportantActivities;
}

export class HideImportantActivitiesAction implements Action {
  public readonly type = ExpertActivitiesPageActionTypes.HideImportantActivities;
}

export class ActivityDetailsClosedAction implements Action {
  public readonly type = ExpertActivitiesPageActionTypes.ActivityDetailsClosed;

  constructor(public payload: GetProfileActivity) {}
}

export class ActivityRowClickAction implements Action {
  public readonly type = ExpertActivitiesPageActionTypes.ActivityRowClick;

  constructor(public payload: { getProfileActivity: GetProfileActivity; isImportant: boolean }) {}
}

export type ExpertActivitiesPageActionUnion =
  | LoadMoreExpertActivitiesAction
  | HideImportantActivitiesAction
  | ShowImportantActivitiesAction
  | ActivityDetailsClosedAction
  | ActivityRowClickAction;
