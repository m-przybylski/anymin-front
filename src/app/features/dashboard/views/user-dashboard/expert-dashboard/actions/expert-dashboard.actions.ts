// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { EmploymentWithService } from '@anymind-ng/api';

export enum ExpertDashboardActionTypes {
  Load = '[Expert Dashboard] Load',
  ReloadAfterEditProfile = '[Expert Dashboard] Reload after edit profile',
  AddConsultation = '[Expert Dashboard] Add consultation',
}

export class LoadExpertDashboardAction implements Action {
  public readonly type = ExpertDashboardActionTypes.Load;

  constructor(public payload: string) {}
}

export class ReloadExpertDashboardAfterEditProfileAction implements Action {
  public readonly type = ExpertDashboardActionTypes.ReloadAfterEditProfile;

  constructor(public payload: string) {}
}

export class AddConsultationAction implements Action {
  public readonly type = ExpertDashboardActionTypes.AddConsultation;
  constructor(public payload: EmploymentWithService) {}
}

export type ExpertDashboardActionUnion =
  | LoadExpertDashboardAction
  | ReloadExpertDashboardAfterEditProfileAction
  | AddConsultationAction;
