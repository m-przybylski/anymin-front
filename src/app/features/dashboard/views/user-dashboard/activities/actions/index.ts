import * as ExpertActivitiesActions from './expert-activities.actions';
import * as ExpertActivitiesApiActions from './expert-activities-api.actions';
import * as ExpertActivitiesPageActions from './expert-activities-page.actions';
import * as BalanceApiActions from './balance-api.actions';
import * as ExpertActivitiesWsActions from './expert-activities-ws.actions';

export {
  ExpertActivitiesActions,
  ExpertActivitiesApiActions,
  ExpertActivitiesPageActions,
  BalanceApiActions,
  ExpertActivitiesWsActions,
};
