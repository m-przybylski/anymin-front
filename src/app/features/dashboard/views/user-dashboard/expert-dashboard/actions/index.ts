import * as ExpertDashboardActions from './expert-dashboard.actions';
import * as ExpertDashboardApiActions from './expert-dashboard.api.actions';

export { ExpertDashboardActions, ExpertDashboardApiActions };
