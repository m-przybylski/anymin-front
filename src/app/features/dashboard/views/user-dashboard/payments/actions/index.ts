import * as PaymentsApiActions from './payments-api.actions';
import * as PaymentsInitActions from './payments-init.actions';

export { PaymentsInitActions, PaymentsApiActions };
