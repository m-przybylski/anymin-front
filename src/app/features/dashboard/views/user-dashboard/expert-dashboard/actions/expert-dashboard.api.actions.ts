// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { IExpertCompanyDashboardResolverData } from '../../../common/resolver-helpers';
import { IExpertProfile } from '../services/expert-dashboard.service';

export enum ExpertDashboardApiActionTypes {
  LoadSuccess = '[Expert Dashboard] Load Success',
  LoadFailure = '[Expert Dashboard] Load Failure',
}

export class LoadExpertDashboardSuccessAction implements Action {
  public readonly type = ExpertDashboardApiActionTypes.LoadSuccess;

  constructor(public payload: IExpertCompanyDashboardResolverData<IExpertProfile>) {}
}

export class LoadExpertDashboardFailureAction implements Action {
  public readonly type = ExpertDashboardApiActionTypes.LoadFailure;

  constructor(public payload: any) {}
}

export type ExpertDashboardActionUnion = LoadExpertDashboardSuccessAction | LoadExpertDashboardFailureAction;
