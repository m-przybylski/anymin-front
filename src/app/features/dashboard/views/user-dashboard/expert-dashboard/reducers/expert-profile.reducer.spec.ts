import { reducer, IState } from './index';
import { ExpertDashboardApiActions, ExpertDashboardActions } from '../actions';
import { ConsultationDetailActions } from '@platform/shared/components/modals/consultation-details/actions';

const initialState: IState = {
  isLoading: false,
};

describe('company profile reducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);

      expect(result).toMatchSnapshot();
    });
  });

  describe('Dupa', () => {
    it('should be Dupa', () => {
      expect('Dupa').toBe('Dupa');
    });
  });

  describe('LoadExpertDashboardAction', () => {
    it('should start loading data', () => {
      const action = new ExpertDashboardActions.LoadExpertDashboardAction('fake profile');
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });
  describe('ReloadExpertDashboardAfterEditProfileAction', () => {
    it('should start reloading data', () => {
      const action = new ExpertDashboardActions.ReloadExpertDashboardAfterEditProfileAction('fake profile');
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });
  describe('LoadExpertDashboardSuccessAction', () => {
    it('should update store with profile data', () => {
      const profile: any = '👍👍👍';
      const action = new ExpertDashboardApiActions.LoadExpertDashboardSuccessAction(profile);
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });
  describe('LoadFailure', () => {
    it('should udpate store with error message', () => {
      const profile: any = '👍👍👍';
      const action = new ExpertDashboardApiActions.LoadExpertDashboardFailureAction(profile);
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });
  describe('AddConsultationAction', () => {
    it('should not modify store when nothing in the store', () => {
      const newConsultation: any = 'fake consultation';
      const action = new ExpertDashboardActions.AddConsultationAction(newConsultation);
      const result = reducer(initialState, action);
      expect(result.expertProfile).toBeFalsy();
    });
    it('should add consultation to the store', () => {
      const state: IState = {
        isLoading: false,
        expertProfile: {
          isCompany: false,
          isLogged: false,
          isOwnProfile: false,
          profile: {
            expertProfileView: {
              employments: ['fake service'] as any,
              expertProfile: {} as any,
              isFavourite: false,
            },
            getProfileWithDocuments: {} as any,
          },
        },
      };
      const newConsultation: any = 'fake consultation';
      const action = new ExpertDashboardActions.AddConsultationAction(newConsultation);
      const result = reducer(state, action);
      expect(result).toMatchSnapshot();
    });
  });
  describe('DeleteConsultationAction', () => {
    it('should delete existing consultation by its id', () => {
      const state: IState = {
        isLoading: false,
        expertProfile: {
          isCompany: false,
          isLogged: false,
          isOwnProfile: false,
          profile: {
            expertProfileView: {
              employments: [
                { serviceDetails: { id: '3' } },
                { serviceDetails: { id: '4' } },
                { serviceDetails: { id: '5' } },
              ] as any,
              expertProfile: {} as any,
              isFavourite: false,
            },
            getProfileWithDocuments: {} as any,
          },
        },
      };
      const consultationToDelete: any = '3';
      const action = new ConsultationDetailActions.DeleteConsultationAction(consultationToDelete);
      const result = reducer(state, action);
      expect(result).toMatchSnapshot();
    });
  });
  describe('EditConsultationAction', () => {
    it('should update existing consultation', () => {
      const state: IState = {
        isLoading: false,
        expertProfile: {
          isCompany: false,
          isLogged: false,
          isOwnProfile: false,
          profile: {
            expertProfileView: {
              employments: [
                { serviceDetails: { id: '3', name: 'old name' } },
                { serviceDetails: { id: '4', name: 'some not valid name' } },
                { serviceDetails: { id: '5', name: 'some not valid name' } },
              ] as any,
              expertProfile: {} as any,
              isFavourite: false,
            },
            getProfileWithDocuments: {} as any,
          },
        },
      };
      const consultationToEdit: any = { id: '3', name: 'new name' };
      const action = new ConsultationDetailActions.EditConsultationAction(consultationToEdit);
      const result = reducer(state, action);
      expect(result).toMatchSnapshot();
    });
  });
});
