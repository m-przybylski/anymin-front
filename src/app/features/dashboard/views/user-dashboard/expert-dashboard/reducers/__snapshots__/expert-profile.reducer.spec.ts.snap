// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`company profile reducer AddConsultationAction should add consultation to the store 1`] = `
Object {
  "expertProfile": Object {
    "isCompany": false,
    "isLogged": false,
    "isOwnProfile": false,
    "profile": Object {
      "expertProfileView": Object {
        "employments": Array [
          "fake service",
          "fake consultation",
        ],
        "expertProfile": Object {},
        "isFavourite": false,
      },
      "getProfileWithDocuments": Object {},
    },
  },
  "isLoading": false,
}
`;

exports[`company profile reducer DeleteConsultationAction should delete existing consultation by its id 1`] = `
Object {
  "expertProfile": Object {
    "isCompany": false,
    "isLogged": false,
    "isOwnProfile": false,
    "profile": Object {
      "expertProfileView": Object {
        "employments": Array [
          Object {
            "serviceDetails": Object {
              "id": "4",
            },
          },
          Object {
            "serviceDetails": Object {
              "id": "5",
            },
          },
        ],
        "expertProfile": Object {},
        "isFavourite": false,
      },
      "getProfileWithDocuments": Object {},
    },
  },
  "isLoading": false,
}
`;

exports[`company profile reducer EditConsultationAction should update existing consultation 1`] = `
Object {
  "expertProfile": Object {
    "isCompany": false,
    "isLogged": false,
    "isOwnProfile": false,
    "profile": Object {
      "expertProfileView": Object {
        "employments": Array [
          Object {
            "serviceDetails": Object {
              "id": "3",
              "name": "new name",
            },
          },
          Object {
            "serviceDetails": Object {
              "id": "4",
              "name": "some not valid name",
            },
          },
          Object {
            "serviceDetails": Object {
              "id": "5",
              "name": "some not valid name",
            },
          },
        ],
        "expertProfile": Object {},
        "isFavourite": false,
      },
      "getProfileWithDocuments": Object {},
    },
  },
  "isLoading": false,
}
`;

exports[`company profile reducer LoadExpertDashboardAction should start loading data 1`] = `
Object {
  "isLoading": true,
}
`;

exports[`company profile reducer LoadExpertDashboardSuccessAction should update store with profile data 1`] = `
Object {
  "error": undefined,
  "expertProfile": "üëçüëçüëç",
  "isLoading": false,
}
`;

exports[`company profile reducer LoadFailure should udpate store with error message 1`] = `
Object {
  "error": "üëçüëçüëç",
  "isLoading": false,
}
`;

exports[`company profile reducer ReloadExpertDashboardAfterEditProfileAction should start reloading data 1`] = `
Object {
  "isLoading": true,
}
`;

exports[`company profile reducer undefined action should return the default state 1`] = `
Object {
  "isLoading": false,
}
`;
